File: planner-prototype/server/src/main/java/kpn/planner/domain/NetworkType.java
package kpn.planner.domain;

public enum NetworkType {
	cycling("rcn"),
	hiking("rwn"),
	horse("rhn"),
	motorboat("rmn"),
	canoe("rpn"),
	inlineSkating("rin");

	private final String tagKey;

	NetworkType(String tagKey) {
		this.tagKey = tagKey;
	}

	public String getTagKey() {
		return tagKey;
	}
}

File: server/src/main/scala/kpn/api/custom/NetworkType.scala
package kpn.api.custom

import enumeratum.Enum
import enumeratum.EnumEntry
import enumeratum.EnumEntry.Hyphencase

sealed trait NetworkType extends EnumEntry with Hyphencase

object NetworkType extends Enum[NetworkType] {

  val values: IndexedSeq[NetworkType] = findValues

  final case object hiking extends NetworkType

  final case object cycling extends NetworkType

  final case object horseRiding extends NetworkType 

  final case object canoe extends NetworkType

  final case object motorboat extends NetworkType

  final case object inlineSkating extends NetworkType
}
